### DISCUSSION TITLE
abnormal soft shadow when apply PCSS shadow type

### DISCUSSION THREAD

**Discussion by cjlong1213** [google/filament#8361] [General]
**Describe the bug**
We are developping avatar project and try to use soft shadow feature, abnormal soft shadow  occurred when apply PCSS shaow type, as shown below:
![Screenshot_2023_1128_161420_](https://github.com/google/filament/assets/15936033/546d036e-f82b-4099-a424-adc56a634234)

**To Reproduce**
1.PCF shadow as below:

![Screenshot_2023_1124_152751](https://github.com/google/filament/assets/15936033/0931cd9d-d30f-49b6-ad8c-ee4e7a364521)
2.Enlarge the intensity of PCF shadow
![Screenshot_2023_1128_160641](https://github.com/google/filament/assets/15936033/080ec080-d4ce-4861-ad12-5a2eb65e1779)
3.Change to PCSS soft shadow type
![Screenshot_2023_1128_161420](https://github.com/google/filament/assets/15936033/82fe016e-8a5d-4be8-bf1b-dc6af443869c)

Any suggestions to fix this? Many thanks!

**Smartphone (please complete the following information):**
 - Device: [vivo X60pro+]
 - OS: [e.g. Android Pie 9.0]

**Comment by pixelflinger** [comment]
PCSS is still work in progress.

**Comment by pixelflinger** [comment]
For your use case you should try VSM shadows + a small blur radius.

**Comment by cjlong1213** [comment]
We change to VSM shadow type but the abnormal shadow still occure, after enlarging the fov from 18.5 degree to 28, the abnormal shadow disappeared. Maybe there is a better way to fix this.
Our use case is mobile phone, as far as I know，VSM shadow has more calculation than others which would consume more power. Is there any suggestion?

**Comment by pixelflinger** [comment]
PCSS is vastly more costly than VSM. So from a performance standpoint, VSM is going to be better. Also since you want a soft result, you can use a very low resolution with VSM, try 64 pixels and use "stable" mode.

Which fov are you taking about? The camera? Can you send pictures of the different cases.

Without a repro case we won't be able to do much.

**Comment by cjlong1213** [comment]
@pixelflinger According to your suggestion, we change the shadow type from PCSS to VSM and set the parameters as you mentioned(64 blur size and "stable" mode), part of the shadow disappears once the blur radius is increased as shown blew:
![image](https://github.com/google/filament/assets/15936033/37a23052-10e5-4d25-807a-829d72dbd598)

Any suggestion?
Thanks in advance.

**Comment by pixelflinger** [comment]
With VSM, the blur radius cannot be too large. You can try to enable `VsmShadowOptions::highPrecision` and/or `ShadowOptions::Vsm::elvsm`. It might help. But regardless, The blur option for VSM cannot go too large.
